scripts/es-tag.sh

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env bash
+
+# Parameters: 
+#  $1 - Fully qualified docker image name e.g. library/elasticsearch
+
+# Do semantic version tagging for elasticsearch/kibana images:
+# 1. Fetch last updated versions
+# 2. Calculate latest major versions
+# 3. Re-tag with semantic versions and push to jahia/ repository
+
+set -eo pipefail
+shopt -s inherit_errexit
+
+# Set VERSIONS_LIST env variable with last updated versions
+fetchVersions() {
+  dockerHubUrl="https://hub.docker.com/v2/repositories/${FULL_IMAGE_NAME}/tags?ordering=last_updated&page_size=50"
+  VERSIONS_LIST=$(curl --get -s "${dockerHubUrl}" \
+    | jq -r '.results[].name' \
+    | sort -Vr)
+}
+
+# Populate MAJOR_VERSIONS dictionary
+# e.g. $MAJOR_VERSIONS[7] => <latest 7.x.x version>
+getLatestVersions() {
+    while read -r ver; do
+      # check MIN_VERSION threshold
+      if [ "$(maxVersion "$ver" "$MIN_VERSION")" == "$ver" ]; then
+        majorTag=$(majorVer "$ver")
+        currMajorVer="${MAJOR_VERSIONS[$majorTag]}"
+        maxVer=$(maxVersion "${ver}" "${currMajorVer}")
+        echo "version:${ver}, majorTag:${majorTag}, maxVersion:$maxVer"
+        MAJOR_VERSIONS+=( ["$majorTag"]="$maxVer" )
+      else
+        echo "skipping version:${ver}"
+      fi
+    done <<< "$VERSIONS_LIST"
+}
+
+tagImages() {
+  for majorTag in "${!MAJOR_VERSIONS[@]}"; do
+    fullTag=${MAJOR_VERSIONS[$majorTag]}
+    echo; docker pull ${FULL_IMAGE_NAME}:${fullTag}
+
+    minorTag="$majorTag.$(minorVer "$fullTag")"
+    tagAndPushImage "${FULL_IMAGE_NAME}":"${fullTag}" jahia/"${IMAGE_NAME}":"${fullTag}"
+    tagAndPushImage "${FULL_IMAGE_NAME}":"${fullTag}" jahia/"${IMAGE_NAME}":"${majorTag}"
+    tagAndPushImage "${FULL_IMAGE_NAME}":"${fullTag}" jahia/"${IMAGE_NAME}":"${minorTag}"
+  done
+}
+
+#+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+#
+
+# Helper methods
+
+# compare and return max version between $1 and $2
+maxVersion() { echo -e "$1\n$2" | sort -Vr | head -n 1; }
+# extract major version e.g. 7.12.4 => 7
+majorVer() { echo -e "$1" | awk -F . '{print $1}'; }
+# extract minor version e.g. 7.12.4 => 12
+minorVer() { echo -e "$1" | awk -F . '{print $2}'; }
+tagAndPushImage() {
+  echo "Tagging $1 => $2..."
+  docker tag "$1" "$2"
+  docker push "$2"
+}
+
+#+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+#
+
+# Variable declaration
+MIN_VERSION=7.0.0 # ignore any version below this
```

Comment:
Since the main purpose of this exercise is to automatically get and refresh a tag to the latest version, we can set the minimum version to today's current release.

Reply:
Adjusted min version to latest 7.17.4